pkg/controller/volume/attachdetach/statusupdater/node_status_updater.go

```diff
@@ -69,10 +69,10 @@ func (nsu *nodeStatusUpdater) UpdateNodeStatuses() error {
 			// reset flag statusUpdateNeeded back to true to indicate this node status
 			// needs to be updated again
 			glog.V(2).Infof(
-				"Could not update node status. Failed to find node %q in NodeInformer cache. %v",
+				"Could not update node status. Failed to find node %q in NodeInformer cache. Error: '%v'",
 				nodeName,
 				err)
-			nsu.actualStateOfWorld.SetNodeStatusUpdateNeeded(nodeName)
+			nsu.actualStateOfWorld.RemoveNodeFromAttachUpdates(nodeName)
```

Comment:
I think you need to take different actions for nodeObj==nil and err != nil
When nodeObj==nil, it shows that API server does not have this object anymore, it should be safe to removeNode. But for err != nil, that indicates something wrong then retrieving the node object and node status updater should try it again.